### ISSUE TITLE
[HOLD for payment 2023-05-05] [$500] Android - Downloaded photo is not appearing in the photo album 

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#11101]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Receive couple photos in the chat 
2. Navigate to the image 
3. Download 
4. locate photo in the album 

## Expected Result:
 if user downloads a photo, it should appear in the photos album

## Actual Result:
When download an image on Android user is not able to locate it in the local file system. User can't find it in the photos album, and it doesn't appear to be in the files app either.

## Workaround:
unknown 

## Platform:

Where is this issue occurring?

- Android

**Version Number:** 1.2.0-0
**Reproducible in staging?:** Y 
**Reproducible in production?:** Y 
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation 
![Screenshot_20220910-121310](https://user-images.githubusercontent.com/43995119/190957595-cb48f47e-d66b-46cc-a4a0-f09543440c3c.png)
![Screenshot_20220910-121106](https://user-images.githubusercontent.com/43995119/190957605-c75a9032-9259-4197-9793-e8dae9e2ed52.png)

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @quinthar 
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1662808002043359 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by jliexpensify** [comment]
Posted:

- Internal: https://www.upwork.com/ab/applicants/1572354024809439232/job-details
- External: https://www.upwork.com/jobs/~010d284b228c354c9b

invited @parasharrajat

**Comment by jliexpensify** [comment]
Bump @Expensify/applause

**Comment by kbecciv** [comment]
@parasharrajat @jliexpensify Issue reproduced in 2 platforms: mWeb/safari and Android app

https://user-images.githubusercontent.com/93399543/192833693-10a00941-4c92-469c-bf51-1fa1d43ce2bb.mp4

https://user-images.githubusercontent.com/93399543/192833679-e68a0793-1e03-4cbf-8dca-fde0c26f93e4.mp4

**Comment by kbecciv** [comment]
@aimane-chnaif You are right, so only Android app has issue.

Uploading RPReplay_Final1664385004.MP4…

**Comment by parasharrajat** [comment]
@aimane-chnaif  Ok, I looked into the proposal https://github.com/Expensify/App/issues/11101#issuecomment-1252361689. Will it create a copy of the same file in external storage?

**Comment by parasharrajat** [comment]
Ok. I don't think we need two copies of a same file.

**Comment by thienlnam** [comment]
Yes that's correct. We should not be saving a file to both internal and external storage

Ideal case:
Downloaded photo -> shows up in photo album
Downloaded file -> shows up in files

**Comment by parasharrajat** [comment]
@aimane-chnaif do you have an updated proposal for not storing the files twice?

**Comment by parasharrajat** [comment]
@jliexpensify Let's double it. 

Waiting for proposals.

**Comment by jliexpensify** [comment]
Doubled - https://expensify.slack.com/archives/C01GTK53T8Q/p1666132117833109

**Comment by parasharrajat** [comment]
I was looking into the Lib methods and It says that we can use the correct path to store the file in external storage directly. Why do you are preferring `MediaStore`? There are some details https://github.com/RonRadtke/react-native-blob-util#android-media-scanner-and-download-manager-support which talks about media scanning. I haven't tested it but will this not work and Why?

And when `addAndroidDownloads` is used, `fileCache: true, path: `${path}/${attachmentName}` config optionsa are ignored.

**Comment by parasharrajat** [comment]
Ok, Thanks. One more thing, Can you please show me a video of the location where files are downloaded? Try with an image and a PDF.

**Comment by parasharrajat** [comment]
Ok, Thanks.  @aimane-chnaif 's [proposal](https://github.com/Expensify/App/issues/11101#issuecomment-1283271264) looks good to me. 

cc: @thienlnam

:ribbon: :eyes: :ribbon:  C+ reviewed

**Comment by jliexpensify** [comment]
Cool, @thienlnam - tag me once you approve and I'll hire. Cheers!

**Comment by thienlnam** [comment]
Thanks for making those proposal updates 🟢 

cc @jliexpensify We're ready for a hire once they apply to the job

**Comment by jliexpensify** [comment]
Hired Rajat and sent an invite to @aimane-chnaif

**Comment by thienlnam** [comment]
PR pending update from @aimane-chnaif - any update here?

**Comment by jliexpensify** [comment]
Bump @aimane-chnaif !

**Comment by parasharrajat** [comment]
Any update?

**Comment by Julesssss** [comment]
Added my thoughts [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1668449157198979?thread_ts=1667776556.029449&cid=C01GTK53T8Q) and on [the issue](https://github.com/Expensify/App/pull/12033#issuecomment-1314172416). I think there is a relatively simple solution here where we just block the DownloadManager notification 🤞

**Comment by Julesssss** [comment]
[On hold ](https://github.com/Expensify/App/pull/12033#issuecomment-1317079741)until a simple notification can be used to replace the DownloadManager notification

**Comment by JmillsExpensify** [comment]
@trjExpensify Maybe you should take this over? You're on the notifications tracking issue. Let's also update the hold title to clarity exactly what issue this is held up on.

**Comment by trjExpensify** [comment]
Gladly! @Julesssss, can we confirm this is still the play? Hold on the notifications design doc to get through to implementation?

**Comment by Julesssss** [comment]
Yep, this will be held until we reach the implementation stage of the Notification polish doc.

**Comment by trjExpensify** [comment]
Still held.

**Comment by JmillsExpensify** [comment]
Leaving the hold for now, though with notifications v1 done, we should consider picking this back up again.

**Comment by thienlnam** [comment]
cc @Julesssss Is this ready to be picked up again?

**Comment by Julesssss** [comment]
The new notification types won't be added until the Notification Polish project, so this isn't quite ready yet

**Comment by JmillsExpensify** [comment]
Assigning myself to this one as well as I'm helping with the notification issues.

**Comment by Julesssss** [comment]
Alright, and [here's](https://github.com/Expensify/App/issues/16044) the follow-up issue for re-introducing the notification for 'Download complete'.

**Comment by trjExpensify** [comment]
@aimane-chnaif any luck? Popping this back on `daily`.

**Comment by Julesssss** [comment]
Seems like the next step is to ask the lib maintainers to verify that their <Andorid 10 fix actually works

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.3.7-3 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/12033


If no regressions arise, payment will be issued on 2023-05-05. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@parasharrajat / @aimane-chnaif] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@parasharrajat / @aimane-chnaif] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@parasharrajat / @aimane-chnaif] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@parasharrajat / @aimane-chnaif] Determine if we should create a regression test for this bug.
- [ ] [@parasharrajat / @aimane-chnaif] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@JmillsExpensify / @trjExpensify] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by trjExpensify** [comment]
>@Julesssss can I request bounty increase based on efforts and time spent?

Jules is out today, but I agree the issue should be on the new $1k base since it came off hold and that change was made, so the $500 title is incorrect for sure.   

Question though, are we totally done here [with this PR](https://github.com/Expensify/App/pull/12033/files) now complete? or are there/will there be follow-on issues for these items that you're taking @aimane-chnaif?

>Instead of hiding download manager notification, show local notification coming from notification project
>Seems like the next step is to ask the lib maintainers to verify that their <Andorid 10 fix actually works. We can offer them money/sponsorship to help us solve the problem.

**Comment by Julesssss** [comment]
Yeah, I think it's fine to close this one out. The follow up not going to be high priority for a while yet.

**Comment by trjExpensify** [comment]
Cool, thanks for confirming the follow-up issue. Offers have been sent for $1,000 on [this new private job](https://www.upwork.com/jobs/~0192f7ccb60fbff642) I needed to create. 

On the [proposed regression test](https://github.com/Expensify/App/issues/11101#issuecomment-1536513559) @parasharrajat, did you look in TestRail? We have [this one](https://expensify.testrail.io/index.php?/cases/view/1977395&group_by=cases:section_id&group_order=asc&display_deleted_cases=0&group_id=229062) for downloading an image (and other) attachments from a chat. 

Seems like what we want to do is expand on those with an Android section there? 

**Android downloads**
1. Open the attachment detail.
2. Download it.
3. Open the native Files app and locate /Download/Expensify.
4. **Verify** that the downloaded file appears in that folder.
5. For media (images or videos) only:
   5a. Open the native Gallery app
   5b. **Verify** that downloaded media appears in the list.

Let me know if you agree with that and I can get this across to Applause.

**Comment by trjExpensify** [comment]
Cool, payments issued and regression test shipped for Applause. Closing!
